tests/testthat/test-AbstractPackageReporter.R

```diff
@@ -0,0 +1,60 @@
+context("Abstract Package Reporter Tests")
+
+##### TEST SET UP #####
+
+rm(list = ls())
+# Configure logger (suppress all logs in testing)
+loggerOptions <- futile.logger::logger.options()
+if (!identical(loggerOptions, list())){
+  origLogThreshold <- loggerOptions[[1]][['threshold']]
+} else {
+  origLogThreshold <- futile.logger::INFO
+}
+futile.logger::flog.threshold(0)
+
+##### TESTS #####
+
+## Structure Available ##
+
+test_that('AbstractPackageReporter structure is as expected', {
+  
+  expect_named(object = AbstractPackageReporter$public_methods
+               , expected = c(
+                 "set_package",
+                 "get_package",
+                 "get_package_path",
+                 "get_report",
+                 "get_summary_view",
+                 "get_raw_data",
+                 "clone"
+               )
+               , info = "Available public methods for AbstractPackageReporter not as expected."
+  )
+  
+  expect_named(object = AbstractPackageReporter$public_fields
+               , expected = NULL
+               , info = "Available public fields for AbstractPackageReporter not as expected."
+  )
+  
+  expect_named(object = AbstractPackageReporter$private_methods
```

Comment:
IMHO we shouldn't be testing private details of these classes. They're private because we want the flexibility to change them. Our tests should only be concerned with the existence and correctness of private methods/members.

I could see a case for testing the _correctness_ of private methods...but I don't think this test makes sense. Like having a test fail if I add an additional private method doesn't help us

Reply:
so (realize I may not have been clear), I would remove the `expect_named` tests on `$private_methods` and `$private_fields`

Reply:
Don't you want to know if a PR is going to change the paradigm of inheritance between the objects?  PRs that do this would just need to acknowledge this by updating the unit test.